### PR TITLE
Add lb traffic extension resource

### PR DESCRIPTION
**Pull request by maxi-cit**
Adds the new google_network_services_lb_traffic_extension resource.

Part of: https://github.com/hashicorp/terraform-provider-google/issues/17491
Fixes: https://github.com/hashicorp/terraform-provider-google/issues/17373

**Release Note Template for Downstream PRs (will be copied)**

```release-note:new-resource
`google_network_services_lb_traffic_extension`
```

### CODE REVIEW

**Review by NickElliot — CHANGES_REQUESTED** [review]
please remove the trailing white spaces on all files, thanks!

**Comment by maxi-cit** [comment]
Hey @NickElliot I am still waiting for the API to become GA in order to validate the tests

PD: they told they are releasing it in the comming weeks

File: mmv1/templates/terraform/examples/network_services_lb_traffic_extension_basic.tf.erb [hunk 1]
```diff
@@ -0,0 +1,217 @@
...
+}
+# [END cloudloadbalancing_int_http_gce]
+
+# [START lb_traffic_extension]
+
+resource "google_network_services_lb_traffic_extension" "<%= ctx[:primary_resource_id] %>" {
+    location   = "us-west1"
+    name       = "lb-traffic-extension"
+    provider      = google-beta
+    forwarding_rules = ["https://www.googleapis.com/compute/v1/projects/myproj/regions/us-west1/forwardingRules/l7-ilb-forwarding-rule"]
```
Comment (NickElliot) [inline mmv1/templates/terraform/examples/network_services_lb_traffic_extension_basic.tf.erb:199]: This needs to be an id reference to a forwarding rule created in this example

File: mmv1/templates/terraform/examples/network_services_lb_traffic_extension_basic.tf.erb [hunk 2]
```diff
@@ -0,0 +1,217 @@
...
+    load_balancing_scheme = "INTERNAL_MANAGED"
+    extension_chains {
+        name = "chain1"
+        match_condition {
+            cel_expression = "request.host == 'example.com'"
+        }
+        extensions {
+            name = "ext11"
+            authority = "ext11.com"
+            service = "https://www.googleapis.com/compute/v1/projects/myproj/regions/us-west1/backendServices/l7-ilb-test-vm"
```
Comment (NickElliot) [inline mmv1/templates/terraform/examples/network_services_lb_traffic_extension_basic.tf.erb:209]: same here but for backend service!

**Review by NickElliot — CHANGES_REQUESTED** [review]
These need to be updated, thanks!

**Comment by maxi-cit** [comment]
Hello @NickElliot, still working on adding integration tests but I should be able to push some of them today for traffic and route extensions

**Comment by NickElliot** [comment]
All good, ping me when you have them ready @maxi-cit

**Comment by NickElliot** [comment]
Also in case it resolves the current VCR error, could you rebase? Thanks!

**Review by NickElliot — COMMENTED** [review]
see comments

**Review by NickElliot — CHANGES_REQUESTED** [review]
``` 
Error: Incorrect attribute value type
Inappropriate value for attribute "supported_events": list of string
```
Sample needs format updated

File: mmv1/templates/terraform/examples/network_services_lb_traffic_extension_basic.tf.erb (hunk 1 above)
Comment (maxi-cit) [reply mmv1/templates/terraform/examples/network_services_lb_traffic_extension_basic.tf.erb:199]: Hello @NickElliot. API is returning the self_link using number, so I do not think we can use the id field here

File: mmv1/products/networkservices/LbTrafficExtension.yaml [hunk 3]
```diff
@@ -0,0 +1,178 @@
...
+      forwarding_rule_name: 'l7-ilb-forwarding-rule'
+      target_http_proxy_name: 'l7-ilb-target-http-proxy'
+      regional_url_map_name: 'l7-ilb-regional-url-map'
+      backend_service_name: 'l7-ilb-backend-subnet'
+      mig_template_name: 'l7-ilb-mig-template'
+      hc_name: 'l7-ilb-hc'
+      mig_name: 'l7-ilb-mig1'
+      fw_allow_iap_hc_name: 'l7-ilb-fw-allow-iap-hc'
+      fw_allow_ilb_to_backends_name: 'l7-ilb-fw-allow-ilb-to-backends'
+      vm_test_name: 'l7-ilb-test-vm'
```
Comment (NickElliot) [inline mmv1/products/networkservices/LbTrafficExtension.yaml:53]:
```suggestion
      vm_test_name: 'l7-ilb-test-vm'
      project: :PROJECT_NAME
```

File: mmv1/templates/terraform/examples/network_services_lb_traffic_extension_basic.tf.erb (hunk 1 above)
Comment (NickElliot) [reply mmv1/templates/terraform/examples/network_services_lb_traffic_extension_basic.tf.erb:199]: the string could be constructed using variables such as 
```"https://www.googleapis.com/compute/v1/projects/<%= ctx[:test_env_vars]['project'] %>/regions/us-west1/forwardingRules/<%= ctx[:vars]['forwarding_rule_name'] %>"```

**Comment by rshriram** [comment]
Any update on this?

**Comment by maxi-cit** [comment]
Hello @NickElliot, is there anything left?

**Review by NickElliot — COMMENTED** [review]
I'm sorry I didn't realize sooner, but in regards to https://github.com/GoogleCloudPlatform/magic-modules/pull/10121#discussion_r1569044945 can you link me the documentation where this is the expected format for a resource reference? It seems very nonstandard that the `https://www.googleapis.com/compute/v1/projects/` header would be required, and want to verify that using the self_link reference does not work as users will expect to use it in that way.

**Review by NickElliot — CHANGES_REQUESTED** [review]
need version guards around the handwritten test file, its failing in the ga provider because the testAccCheckNetworkServicesLbTrafficExtensionDestroyProducer comes from the generated test file, which is beta only

**Comment by maxi-cit** [comment]
Hello @NickElliot, I removed the beta annotation from the mmv1 specification. According to this [doc](https://cloud.google.com/service-extensions/docs/reference/rest/v1/projects.locations.lbTrafficExtensions/create). This is also available in GA.

File: mmv1/products/networkservices/LbTrafficExtension.yaml [hunk 4]
```diff
@@ -0,0 +1,186 @@
...
+update_mask: true
+autogen_async: true
+async: !ruby/object:Api::OpAsync
+  operation: !ruby/object:Api::OpAsync::Operation
+    base_url: '{{op_id}}'
+examples:
+  - !ruby/object:Provider::Terraform::Examples
+    name: 'network_services_lb_traffic_extension_basic'
+    primary_resource_id: 'default'
+    min_version: beta
```
[the remark this answers is not in the document]
Comment (maxi-cit) [reply mmv1/products/networkservices/LbTrafficExtension.yaml]: I removed that attribute

**Review by NickElliot — APPROVED** [review]
LGTM

**Pull request merged**